### ISSUE TITLE
Coordinator can't stop in time, because of background jobs is still running

### ISSUE THREAD

**Issue by in-han** [tikv/pd#5274]
## Bug Report
I have posted a post on the forum,   see  https://asktug.com/t/topic/694191/3 .
In short, the problem is that when coordinator is stopping the scheduler will still keep running until  all jobs are finished.

The following figure shows that it takes a long time for the coordinator to close from the beginning to the end,  in this case more than 10 minutes.
<img width="1057" alt="image" src="https://user-images.githubusercontent.com/4653075/177483723-25a3f343-88c4-4294-834b-8193984cb8f2.png">

### What did you do?
pd leader lease timeout.
<img width="946" alt="image" src="https://user-images.githubusercontent.com/4653075/177484108-3a393e42-ffa2-465b-926c-287890d49f5b.png">

### What did you expect to see?
Follower campaign for leader in time.

### What did you see instead?
Follower can't campaign for leader in 10 minutes, and keep print the following log.
<img width="803" alt="image" src="https://user-images.githubusercontent.com/4653075/177485589-baed9020-d47f-4eab-bb6a-ba2be3924947.png">

### What version of PD are you using (`pd-server -V`)?
5.3.0

**Comment by nolouch** [comment]
Thanks, @in-han , I think the coordinator running too long does not affect the follower who became a leader, the follower should watch the leader key is expired. could you show more details logs?

**Comment by in-han** [comment]
> Thanks, @in-han , I think the coordinator running too long does not affect the follower who became a leader, the follower should watch the leader key is expired. could you show more details logs?

Yes, follower can watch key expiration.  But follower can campaign leader with tow conditions:  a) pd leader key is expired;  b) follower is etcd leader, in this case, the old pd leader was still the etcd leader.

This is the logs:
<img width="1203" alt="image" src="https://user-images.githubusercontent.com/4653075/177534821-1469dea2-7c46-4587-992b-ce246cd0034c.png">

**Comment by nolouch** [comment]
Why lease expired but the old PD still is the leader? does it re-election and the old pd became leader again?

BTW, 2000 TiKVs is the largest cluster size I have seen. amazing! and here is an issue trace to improve the performance of the `balance region` https://github.com/tikv/pd/issues/3744. If you have an interest, may you can help us.

**Comment by in-han** [comment]
Pd leader lease timeout because it encountered timeout when write key to embed ETCD.  You are write, in this case the old pd became leader again.

Ah, after deployment, we also feel that this cluster is too large!

I think there maybe two ways to resolve this problem :  
a) speed up coordinator stop. [There is already a PR](https://github.com/tikv/pd/pull/4148) had optimized this problem ,but it wasn't enough. I have some ideas and will try to make a PR.   
b) as you mentioned, optimize scheduler performance, I will have a look at the PR.

**Comment by in-han** [comment]
@nolouch Thanks for your reply
